cluster-operator/src/test/java/io/strimzi/operator/cluster/ClusterOperatorTest.java

```diff
@@ -165,29 +169,37 @@ private void startStop(VertxTestContext context, String namespaces, boolean open
 
         Map<String, String> env = buildEnv(namespaces);
 
-        Checkpoint async = context.checkpoint();
+        CountDownLatch latch = new CountDownLatch(namespaceList.size() + 1);
+
         Main.run(vertx, client, new PlatformFeaturesAvailability(openShift, KubernetesVersion.V1_9),
                     ClusterOperatorConfig.fromMap(env, KafkaVersionTestUtils.getKafkaVersionLookup()))
             .onComplete(context.succeeding(v -> context.verify(() -> {
                 assertThat("A verticle per namespace", vertx.deploymentIDs(), hasSize(namespaceList.size()));
-
                 for (String deploymentId: vertx.deploymentIDs()) {
-                    vertx.undeploy(deploymentId, context.succeeding());
+                    vertx.undeploy(deploymentId, asyncResult -> {
+                        if (asyncResult.failed()) {
+                            log.error("Failed to undeploy {}", deploymentId);
+                            context.failNow(asyncResult.cause());
+                        }
+                        latch.countDown();
+                    });
                 }
 
                 int maximumExpectedNumberOfWatchers = (openShift ? 9 : 7) * namespaceList.size(); // we do not have connectS2I on k8s
                 assertThat("Looks like there were more watchers than namespaces",
                         numWatchers.get(), lessThanOrEqualTo(maximumExpectedNumberOfWatchers));
-                async.flag();
+                latch.countDown();
             })));
+        latch.await(10, TimeUnit.SECONDS);
+        context.completeNow();
```

Comment:
I don't really understand why this had to change due to a simple change in JDK. If this is just trying to remove a flaky test perhaps that should be done in a separate PR, or at least the PR description should mention that you've fixed a flaky test in addition to changing the JDK.

Reply:
With java11 it started failing more often so most of the java11 builds failed. I wanted to make the build green. I can split it into another PR if you wish.